src/framework/global/types/bytearray.cpp

```diff
@@ -128,6 +128,14 @@ bool ByteArray::operator==(const ByteArray& other) const
     return std::memcmp(constData(), other.constData(), size()) == 0;
 }
 
+bool ByteArray::operator<(const ByteArray& other) const
+{
+    if (size() < other.size()) {
+        return true;
+    }
+    return std::memcmp(constData(), other.constData(), size()) < 0;
```

Comment:
If this line gets reached, it means that `size() >= other.size()`. In particular, if `size() > other.size()`, then this code causes reading beyond the bounds of `other`. Or am I mistaken?

Also, this implementation can have strange effects. Consider `b1 = "234"` and `b2 = "1234"`. If you call `b1 < b2`, it will return true, because `b1.size() < b2.size()`. But if you call `b2 < b1`, that will _also_ return true (at least if you're lucky and it doesn't crash), because `std::memcmp(b2.constData(), b1.constData(), b2.size()) < 0`.

Reply:
Nice catch! Thank you for pointing that out to me.

Reply:
FWIW, my guess would be:
```c++
// Check the common number of bytes lexicographically
int cmp = std::memcmp(constData(), other.constData(), std::min(size(), other.size));
if (cmp < 0) {
    return true;
}

if (cmp == 0) {
    // If the common number of bytes are equal, the shortest one comes first
    return size() < other.size();
}

// else, the common number of bytes are not in lexicographical order, so return false
return false;
```

Reply:
@cbjeukendrup  Your example works, but looks like this is more straight forward solution.
```cpp
    if (size() < other.size()) {
        return true;
    } else if (size() > other.size()) {
        return false;
    }
    return std::memcmp(constData(), other.constData(), size()) < 0;
```
No need to compare them if one bigger than other

Reply:
But that would mean that `"456" < "12345"`, which is a bit strange.

Reply:
Your right, so your first example is the correct one.